backend/routes_bulk_import.py

```
@@ -0,0 +1,347 @@
+                    file.seek(0)
+                    csv_content = file.read().decode('latin-1')
+                except Exception as e:
+                    logger.error(f"Error reading CSV file: {str(e)}")
+                    return jsonify({'error': 'Unable to read CSV file. Please ensure it is properly encoded.'}), 400
+            
+            if not csv_content.strip():
+                return jsonify({'error': 'CSV file is empty'}), 400
+            
+            # Perform validation only (no database operations)
+            if import_type == 'tools':
+                from utils.bulk_import import parse_csv_content, validate_tool_data, check_duplicate_tool
```

Flake8 (7.2.0)

[error] line too long (105 > 79 characters)

(E501)